app/controllers/application_controller.rb

```diff
@@ -20,28 +19,52 @@ def jsonapi_render(collection, filters)
   end
 
   def set_user
-    @current_user = if auth_token.present?
-        User.from_jwt(Authenticator.from_token(auth_token))
-      else
-        nil
-      end
+    return nil unless auth_token.present?
+
+    @current_user = User.where(
+                      id: client.authenticated_user_id,
+                      login: client.authenticated_user_login
+                    ).first_or_create do |user|
+                      user.display_name = client.authenticated_user_display_name
+
+                      # Explicitly set user admin accessor if encoded in JWT
+                      user.admin = client.authenticated_admin?
```

Comment:
How will a user that is no longer and admin in the API (say staff have left) have the admin attribute here updated?

Reply:
Is there a need to track last sign dates in etc?

Reply:
Ah, that must have been why I had the `.tap` on the end. I was trying to figure out why I'd need to update display_name, but double checking admin is probably worth the millisecond it'll take to run the block every time.